File: LocalPods/DialogView/DialogView/Basic/Element/BasicDialogElement.m
//
//  BasicDialogElement.m
//  TCLPlus
//
//  Created by OwenChen on 2020/8/17.
//  Copyright © 2020 TCL Electronics Holdings Ltd. All rights reserved.
//

#import "BasicDialogElement.h"


@implementation BasicDialogElement

- (void)setModel:(BasicDialogModel *)model {
    _model = model;

    NSString *bgColor = _model.bgColor.length > 0 ? _model.bgColor : @"#FFFFFF";
    self.backgroundColor = [UIColor dialog_colorWithHexString:bgColor];
    self.layer.cornerRadius = _model.cornerRadius;
    self.clipsToBounds = YES;
}

// 需要子类重写
+ (CGFloat)elementHeightWithModel:(BasicDialogModel *)model elementWidth:(CGFloat)elementWidth {
    return 0;
}

@end

File: LocalPods/DialogViewSwift/DialogViewSwift/Basic/Elements/BasicDialogElement.swift
//
//  BasicDialogElement.swift
//  TCLHome
//
//  Created by huaizhang.chen on 2022/2/18.
//  Copyright © 2022 TCL Eagle Lab. All rights reserved.
//

import Foundation
import UIKit

class BasicDialogElement: UIView {
    var model: BasicDialogModel? {
        didSet {
            setUpDetailDialogElement()
        }
    }

    func setUpDetailDialogElement() {
        if let unwrappedModel = model {
            backgroundColor = UIColor(hexString: unwrappedModel.backColor)
            layer.cornerRadius = unwrappedModel.cornerRadius
            clipsToBounds = true
        }
    }

    class func elementHeight(model: BasicDialogModel, elementWidth: CGFloat) -> CGFloat {
        return 0.0
    }
}
